File: tests/syntax.rb
# Some commentary
class Animal
    attr_accessor :name, :type

    def initialize(name, type)
        @name = name
        @type = type
    end

    def make_sound
        puts "#{@name} makes a sound."
    end
end

def greet(animal)
    "Hello, #{animal.name}!"
end

dog = Animal.new("Rex", "dog")
dog.make_sound
puts greet(dog)

=begin
  Some longer commentary
=end
numbers = [1, 2, 3, 4, 5]
squared = numbers.map { |num| num ** 2 }
puts "Squared numbers: #{squared.inspect}"

begin
    raise "An error occurred!"
rescue => e
    puts "Caught exception: #{e.message}"
end

File: tests/syntax.java
// Some commentary
class Animal {
    private String name;
    private String type;

    public Animal(String name, String type) {
        this.name = name;
        this.type = type;
    }

    public void makeSound() {
        switch (type) {
            case "dog":
                System.out.println(name + " barks!");
                break;
            case "cat":
                System.out.println(name + " meows!");
                break;
            default:
                System.out.println(name + " makes a sound.");
        }
    }
}

/*
 * Some longer commentary
 */
public class Main {
    public static void main(String[] args) {
        Animal dog = new Animal("Rex", "dog");
        dog.makeSound();

        try {
            int result = 10 / 0;
        } catch (ArithmeticException e) {
            System.out.println("Caught exception: " + e.getMessage());
        }
    }
}

